packages/grafana-data/src/types/datasource.ts

```diff
@@ -323,6 +323,13 @@ abstract class DataSourceApi<
    */
   modifyQuery?(query: TQuery, action: QueryFixAction): TQuery;
 
+  /**
+   * Used in explore for Log details
+   *
+   * @alpha
+   */
+  analyzeQuery?(query: TQuery, options: AnalyzeQueryOptions): boolean;
```

Comment:
I think @ifrost suggested the same. I think it makes sense, but I would also have to move `modifyQuery` there because both things are part of the same feature.

Reply:
Done, through the addition of `DataSourceWithQueryManipulationSupport`.